### ISSUE TITLE
[BUG] .NET9 SkiaSharp.Views.Blazor my app won't start

### ISSUE THREAD

**Issue by Frostchi** [mono/SkiaSharp#3068]
### Description

After upgrading my previously working .NET 8 app to .NET 9 using Blazor WASM, it fails to run in both Debug/Release builds.

### Code

```html
    <div class="canvas-wrapper">
        <SkiaSharp.Views.Blazor.SKCanvasView OnPaintSurface="OnPaintSurface" class="skia-canvas" EnableRenderLoop="true" />
    </div>
```

### Expected Behavior

App should start.

### Actual Behavior

App fails to start due to unhandled exception:

System.DllNotFoundException: libSkiaSharp
   at SkiaSharp.SKData..cctor()

The page loading progress bar stays stuck at 100% and my Page won't load at all.

### Version of SkiaSharp

2.88.9 (Current)

### Last Known Good Version of SkiaSharp

Other (Please indicate in the description)

### IDE / Editor

Visual Studio (Windows)

### Platform / Operating System

Windows

### Platform / Operating System Version

Windows/Browser

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by imerzan** [comment]
I also tried `3.118.0-preview.1.2` and had the same issue.

```cs
Unhandled Exception:
System.DllNotFoundException: libSkiaSharp
   at SkiaSharp.SKData..cctor()
```

**Comment by capdiem** [comment]
I also have this issue using 3.116.0-preview.6.1
```cs
var imageInfo = new SKImageInfo(...)
```
```
System.TypeInitializationException: The type initializer for 'SkiaSharp.SKImageInfo' threw an exception.
 ---> System.DllNotFoundException: libSkiaSharp
   at SkiaSharp.SKImageInfo..cctor()
```

**Comment by TRSkillman** [comment]
This is quite a serious issue for us too as we couldn't get 2.88.9 or 3.118.0-preview.1.2 working in .NET 9 using Blazor WASM.  
We had to roll-back to .NET 8.  
I hope this gets fixed soon!

**Comment by jahnotto** [comment]
I don't know if it's related, but fo-dicom has some issues with loading plugins because .NET 9 has issues figuring out which architecture the CPU is. I don't know any more details -- but perhaps that's what's happening here as well as the dll can't be loaded.

It would be good to be able to upgrade to .NET 9 for sure.

**Comment by LucaCris** [comment]
A free sunday morning to upgrade my Blazor/SkiaSharp project to NET9 and... DllNotFoundException: libSkiaSharp 💩

**Comment by thomasstockermc** [comment]
This pull request should fix this. The underlying issue is a Bug in the .NET 9 Framework.

https://github.com/mono/SkiaSharp/pull/3064

**Comment by mattleibow** [comment]
This may be fixed with https://github.com/mono/SkiaSharp/pull/3064

Can you try the nightly on the preview feed: `3.118.0-nightly.27`

```
https://aka.ms/skiasharp-eap/index.json
```

Let me know if this fixes it and I will push out the 3.x release.

**Comment by capdiem** [comment]
@mattleibow `3.118.0-nightly.27` is working, thanks!

**Comment by mattleibow** [comment]
I am hoping that I can release this week.

**Comment by jahnotto** [comment]
> I am hoping that I can release this week.

Awesome. Thank you!

**Comment by manu-aesim** [comment]
> @mattleibow `3.118.0-nightly.27` is working, thanks! When will version 3.x be released?

I had a similar issue, and this nightly build resolved it for me as well. Thanks!

**Comment by 18779758522** [comment]
> This may be fixed with #3064
> 
> Can you try the nightly on the preview feed: `3.118.0-nightly.27`
> 
> ```
> https://aka.ms/skiasharp-eap/index.json
> ```
> 
> Let me know if this fixes it and I will push out the 3.x release.

Using version 3.118.0-nightly.27, an issue was found. If there is a BasePath path, the SKHtmlCanvas file will not be loaded.Error 404 is reported
Is it related to here?   "/" + moduleUrl
`public JSModuleInterop(IJSRuntime js, string moduleName, string moduleUrl)
		{
#if NET7_0_OR_GREATER
			moduleTask = JSHost.ImportAsync(moduleName, "/" + moduleUrl);
#else
			if (js is not IJSInProcessRuntime)
				throw new NotSupportedException("SkiaSharp currently only works on Web Assembly.");

			moduleTask = js.InvokeAsync<JSObject>("import", moduleUrl).AsTask();
#endif
		}`

**Comment by 18779758522** [comment]
![a84a031f-8bce-4d18-a0d4-cc72fafa9e9e](https://github.com/user-attachments/assets/d411e9f5-f4de-4465-b78c-1271b71d832d)
![a08b628a-037b-45fc-8427-9f175864d834](https://github.com/user-attachments/assets/5f912509-1467-4e0b-80ec-48f580a63609)

**Comment by mattleibow** [comment]
@18779758522 can you add a sample so I can investigate?

**Comment by mattleibow** [comment]
@18779758522 this PR should fix it for you: https://github.com/mono/SkiaSharp/pull/3092

**Comment by mattleibow** [comment]
Closing this as all the issues mentioned in here have been fixed by one PR or another. Hopefully once CI is finished I can release. It has been a long time.

**Comment by 18779758522** [comment]
> @18779758522这个PR应该可以修复它：#3092



> @18779758522 this PR should fix it for you: #3092

Is there a release preview? I'd love to test it out

**Comment by mattleibow** [comment]
This should be on nuget.org now

**Comment by harveytriana** [comment]
Update to 3.118.0-preview.2.3

Start Without Debbuging: OK!
Start With Debbuging: *No canvas element was provided.*

Minimal sample: https://github.com/harveytriana/BlazorMinimalSS9

**Comment by mattleibow** [comment]
@harveytriana thanks for letting me know, can you try 3.116.0?

Also, please open a new issue with your findings. Thanks!

**Comment by LucaCris** [comment]
I can confirm the "random appearing" canvas.

**Comment by markushaslinger** [comment]
I got `No canvas element was provided` after upgrading until I cleared the browser cache (in Firefox the one that does _not_ go away with Ctrl+F5, but has to be deleted in `about:preferences#privacy > Clear Data`) and it worked since then, debug or not.
If you guys can repro it then there must be another error somewhere, but maybe this helps track it down.

**Comment by mattleibow** [comment]
Hmmm, I wonder if I need to make the js files not cached. I thought it was, but I guess not.

**Comment by xela-trawets** [comment]
In ScottPlot 
under dotnet 8, ScottPlot using  <PackageReference Include="SkiaSharp.Views.Blazor" Version="3.118.0-preview.2.3" />
@harveytriana HarveyTriana's observation holds in the blazor demo code - also mouse events are disconnected- interop maybe?
[https://github.com/ScottPlot/ScottPlot/issues/4584](url)


> Update to 3.118.0-preview.2.3
> Start Without Debbuging: OK! Start With Debbuging: _No canvas element was provided._
> Minimal sample: https://github.com/harveytriana/BlazorMinimalSS9
